Adds a warning about manually mounting snapshots managed by ILM

kosabogi opened the pull request:
## Overview

This PR introduces a warning to the documentation for the Mount Snapshot API. 

**Related issue:** [#105816 ](https://github.com/elastic/elasticsearch/issues/105816)

### Preview

[Searchable snapshots](https://elasticsearch_bk_111883.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/searchable-snapshots.html)
[Mount snapshot API](https://elasticsearch_bk_111883.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/searchable-snapshots-api-mount-snapshot.html)

kosabogi committed:
Adds a warning about manually mounting snapshots managed by ILM

elasticsearchmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

DaveCTurner reviewed (dismissed):
I'd rather we didn't add warnings like this to the top of a page. They tend to accumulate, and it makes the docs very hostile to users who are looking for information about how to use the system rather than how not to do so.

Would it make sense to add this information to the other warnings on the page about searchable snapshots: https://www.elastic.co/guide/en/elasticsearch/reference/current/searchable-snapshots.html#searchable-snapshots-reliability

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

shainaraskas reviewed (commented):
quick drive-by review with some formatting and style recos :)

kosabogi committed:
Shortens text and moves the warning to Searchable snapshots chapter

kosabogi commented:
@DaveCTurner Thank you for the feedback! I’ve moved the warning to the section about Searchable Snapshots, where it complements the existing information. In the Mount API description, I’ve added a brief note that links to the warning for further details.

kosabogi commented on docs/reference/searchable-snapshots/apis/mount-snapshot.asciidoc:
```diff
@@ -7,6 +7,19 @@
+
+* **Best Practice:** It's best to let ILM manage your snapshots. If you must manually mount a snapshot, make sure you understand how this affects ILM and manage the snapshot lifecycle separately.
+
+Always review the documentation and consider the impact on your data management before manually handling snapshots managed by ILM.
```
Thank you for the feedback! 
I’ve shortened the warning to make it clearer and fixed the formatting issues. I encountered some difficulties with the local build but have resolved those issues now thanks to @szabosteve :)
Based on Dave's feedback, I’ve moved the warning to the section about Searchable Snapshots, where it complements the existing information. In the Mount API description, I’ve added a brief note that links to the warning for further details.

szabosteve reviewed (approved):
LGTM! 🚢

shainaraskas commented on docs/reference/searchable-snapshots/apis/mount-snapshot.asciidoc:
```diff
@@ -23,6 +23,8 @@ For more information, see <<security-privileges>>.
 ==== {api-description-title}
 
+This API mounts a snapshot as a searchable snapshot index. 
+Note that manually mounting {ilm-init}-managed snapshots can <<manually-mounting-snapshots,interfere>> with <<index-lifecycle-management,{ilm-init} processes>>.
```
I'd drop this in as a separate paragraph so it's more visible. Also, avoid "note that" - either it should be a callout, or it's just a sentence. technically all documentation is just noting things.

I think that because this can cause problems, and we want to discourage people from using this API in the context of ILM-managed snapshots, I'd say that a little more clearly as well.

```suggestion

Don't use this API for snapshots managed by {ilm-init}. Manually mounting {ilm-init}-managed snapshots can <<manually-mounting-snapshots,interfere>> with <<index-lifecycle-management,{ilm-init} processes>>.
```

shainaraskas commented on docs/reference/searchable-snapshots/index.asciidoc:
```diff
@@ -170,6 +170,17 @@ do not have a dedicated frozen tier, you must configure the
+[[manually-mounting-snapshots]]
+[WARNING]
+.Manual snapshot mounting
+====
+Manually mounting snapshots captured by an Index Lifecycle Management ({ilm-init}) policy can
+interfere with {ilm-init}'s automatic management. This may lead to issues such as data loss
+or complications with snapshot handling. For optimal results, allow {ilm-init} to manage
+snapshots automatically. If manual mounting is necessary, be aware of its potential
+impact on {ilm-init} processes. For more information, learn about <<index-lifecycle-management,{ilm-init} snapshot management>>.
+====
```
consider breaking this into a couple of paragraphs. I'm also unclear on whether the complications you mentioned in your last draft ("can disrupt ILM actions, like deletions or phase transitions") is totally covered by this statement.

I'd avoid restating the problem, as you do in the "if you need to mount manually" sentence. it seems to imply that there are impacts not listed here. if it is meant to imply that, we should probably list the additional impacts as I don't think they're captured elsewhere.

finally, I don't think your link is ideal - there's no snapshot info on that top level ILM page. Consider a different link, or changing your link text to refer to just ILM.

```suggestion
====
Manually mounting snapshots captured by an Index Lifecycle Management ({ilm-init}) policy can
interfere with {ilm-init}'s automatic snapshot management. This may lead to issues such as data loss
or complications with snapshot handling.

For optimal results, allow {ilm-init} to manage
snapshots automatically. 

<<index-lifecycle-management,Learn more about {ilm-init} snapshot management>>.
====
```

shainaraskas reviewed (commented):
couple of comments for your consideration!
